File: rules/vuln_samples/positive/javascript/command_injection.js
// POSITIVE: Command injection
const { exec, execSync } = require('child_process');

function runCommand(req) {
    const cmd = req.query.cmd;
    // Unsafe: template literal in exec
    exec(`ls ${cmd}`);
    execSync(`ping ${req.body.host}`);
    // Unsafe: concatenation in exec
    exec("cat " + req.params.filename);
}

File: rules/vuln_samples/positive/python/command_injection.py
# POSITIVE: Command injection — should trigger py-command-injection
import os
import subprocess
from flask import request

def run_command():
    user_input = request.args.get('cmd')
    
    # Unsafe: f-string in os.system
    os.system(f"ls {user_input}")
    # Unsafe: concatenation in subprocess
    subprocess.call("ping " + user_input, shell=True)
    # Unsafe: popen
    os.popen(f"cat {user_input}")
